### PR TITLE
TASK-6345 - Cohort and sample double association is missing

### PR DESCRIPTION
**Pull request by pfurio**

### CODE REVIEW

**Comment by halender** [comment]
Task linked: [TASK-6345 Cohort and sample double association is missing](https://app.clickup.com/t/36631768/TASK-6345)

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/CohortMongoDBAdaptor.java [hunk 1]
```diff
@@ -235,22 +235,21 @@ public OpenCGAResult update(long cohortId, ObjectMap parameters, QueryOptions qu
...
+                if (documentResult.getNumResults() == 0) {
+                    throw new CatalogDBException("Could not update cohort. Cohort uid '" + cohortUid + "' not found.");
+                }
+                return transactionalUpdate(clientSession, documentResult.first(), parameters, variableSetList, queryOptions);
+            });
         } catch (CatalogDBException e) {
-            logger.error("Could not update cohort {}: {}", cohortId, e.getMessage(), e);
-            throw new CatalogDBException("Could not update cohort " + cohortId + ": " + e.getMessage(), e.getCause());
+            logger.error("Could not update cohort {}: {}", cohortUid, e.getMessage(), e);
+            throw new CatalogDBException("Could not update cohort " + cohortUid + ": " + e.getMessage(), e.getCause());
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/CohortMongoDBAdaptor.java:252]: Why are you forwarding just the cause and not the whole exception?
Wouldn't you prefer to catch any exception here?

Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/CohortMongoDBAdaptor.java]: Changed

**Review by j-coll — DISMISSED** [review]

File: opencga-app/src/main/java/org/opencb/opencga/app/migrations/v2_12_6/syncCohortsAndSamplesMigration.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.stream.Collectors;
+
+@Migration(id = "syncCohortsAndSamplesMigration" ,
+        description = "Sync array of samples from cohort with array of cohortIds from Sample",
+        version = "2.12.6",
+        domain = Migration.MigrationDomain.CATALOG,
+        language = Migration.MigrationLanguage.JAVA,
+        date = 20240621
+)
+public class syncCohortsAndSamplesMigration extends MigrationTool {
```
Comment (j-coll) [inline opencga-app/src/main/java/org/opencb/opencga/app/migrations/v2_12_6/syncCohortsAndSamplesMigration.java:27]: Why lower case? didn't the checkstyle complain on this? :o

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/CohortMongoDBAdaptor.java [hunk 3]
```diff
@@ -235,22 +235,21 @@ public OpenCGAResult update(long cohortId, ObjectMap parameters, QueryOptions qu
...
+                        Arrays.asList(QueryParams.ID.key(), QueryParams.UID.key(), QueryParams.STUDY_UID.key(),
+                                QueryParams.SAMPLES.key() + "." + QueryParams.ID.key()));
+                OpenCGAResult<Cohort> documentResult = get(clientSession, query, options);
+                if (documentResult.getNumResults() == 0) {
+                    throw new CatalogDBException("Could not update cohort. Cohort uid '" + cohortUid + "' not found.");
+                }
+                return transactionalUpdate(clientSession, documentResult.first(), parameters, variableSetList, queryOptions);
+            });
+        } catch (Exception e) {
+            logger.error("Could not update cohort {}: {}", cohortUid, e.getMessage(), e);
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/CohortMongoDBAdaptor.java:251]: Aren't we auditing cohort update errors?

**Review by j-coll — APPROVED** [review]

File: opencga-app/src/main/java/org/opencb/opencga/app/migrations/v2_12_6/syncCohortsAndSamplesMigration.java (hunk 2 above)
Comment (pfurio) [reply opencga-app/src/main/java/org/opencb/opencga/app/migrations/v2_12_6/syncCohortsAndSamplesMigration.java]: You can definitely complain. I just changed it :P

**Review by j-coll — APPROVED** [review]

**Pull request merged**